modules/gatsby/src/templates/doc-page-markdown.jsx

```diff
@@ -20,39 +58,35 @@ export const query = graphql`
   }
 `;
 
-function replaceLinks(props) {
-  const {html} = props.data.docBySlug;
-  const {relativeLinks} = props.pageContext;
-
-  return html.replace(/href="([^"]+)"/g, (link, href) => {
-    // don't rewrite external links, don't rewrite links to anchors
-    if (href.startsWith('http') || href.startsWith('#')) {
-      // TODO - we could style them differently though
-      return link;
-    }
-    const hrefWithoutLeadingSlash = href.startsWith('/') ? href.slice(1) : href;
-    // replace links to:
-    // - known physical files, either relative to this file or relative to root
-    // - known routes, either relative to the route of this page or to the home page
-    // by a link to their corresponding route, expresed relative to the home page
-    return `href="${relativeLinks[hrefWithoutLeadingSlash]}"`;
-  });
-}
-
 export default class DocTemplate extends React.Component {
   constructor(props) {
     super(props);
-    this.state = {html: replaceLinks(props)};
+    const {relativeLinks} = props.pageContext;
+    // note - we can add many other custom components.
+    const renderAst = new rehypeReact({
```

Comment:
Care to add a brief comment about what rehypeReact does here? I am not familiar with the component